libs/common/src/types/any-object.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * This type is intended to represent any generic javascript
+ * object. This type is distinct from the `object` type.
+ *
+ * Many types in DTFS (specifically db models stored in the
+ * Mongo database) are not fully documented. Typically, the
+ * only thing we know for sure about these types is that they
+ * will be javascript objects. We normally use the `object`
+ * type in such cases, but the `object` type actually means
+ * "any type which is not `unknown` or `undefined`", which
+ * does not actually capture what we are intending to. The
+ * below type overcomes this issue, providing an alias that
+ * can be used in such cases where we know that the type
+ * will be "any 'javascript' object".
+ *
+ * @example
+ * const foo: object = { value: 100 };
+ * foo['value']; // Property 'value' does not exist on type '{}'
+ *
+ * const bar: AnyObject = { value: 100 };
+ * bar['value']; // works as expected, type = unknown
+ * bar.value; // even this is now fine, type = unknown
+ */
+export type AnyObject = Record<string, unknown>;
```

Comment:
You could use typescripts built-in [PropertyKey](https://www.totaltypescript.com/concepts/propertykey-type) here instead of `string`.

Reply:
That's a cool type I didn't know about. In this specific case, I think the references to this kind of `AnyObject` in DTFS are specifically for things with `string` keys specifically, but it's nice to know that there does exist a built-in type for "anything which can be used as a key"